### PR TITLE
[KOGITO-3297] Bump CloudEvents api to 2.x

### PR DESCRIPTION
**Pull request by ruromero**
Related to https://github.com/kiegroup/kogito-runtimes/pull/750
Fix https://issues.redhat.com/browse/KOGITO-3297

Signed-off-by: ruromero <rromerom@redhat.com>

### CODE REVIEW

**Review by ricardozanini — APPROVED** [review]

**Review by evacchi — APPROVED** [review]

**Comment by ruromero** [comment]
jenkins retest this

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/messaging/BaseEventConsumer.java [hunk 1]
```diff
@@ -50,16 +49,14 @@ public BaseEventConsumer(final TrustyService service) {
...
         } catch (IllegalStateException e) {
             LOG.error(String.format("Can't decode message to CloudEvent: %s", payload), e);
             return Optional.empty();
         }
     }
 
-    protected abstract TypeReference<CloudEventImpl<E>> getCloudEventType();
-
-    protected abstract void handleCloudEvent(final CloudEventImpl<E> cloudEvent);
+    protected abstract void handleCloudEvent(final CloudEvent cloudEvent);
```
Comment (danielezonca) [inline trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/messaging/BaseEventConsumer.java]: What about refactor `getCloudEventType()` to return `TypeReference<E>`?
`ObjectMapper` can be specified as constructor parameter and then you can move deserialization logic here and isolate business logic for subclasses like:
```java
    protected abstract void internalHandleCloudEvent(final CloudEvent cloudEvent, final E payload);

    protected abstract void handleCloudEvent(final CloudEvent cloudEvent) {
        final E payload;
        try {
            payload = mapper.readValue(cloudEvent.getData(), getCloudEventType());
        } catch (IOException e) {
            LOG.error("Unable to deserialize CloudEvent data as " + getCloudEventType().getName(), e);
            return;
        }
        if (event == null) {
            LOG.error("Received CloudEvent with id {} from {} with empty data", cloudEvent.getId(), cloudEvent.getSource());
            return;
        }

        LOG.debug("Received CloudEvent with id {} from {}", cloudEvent.getId(), cloudEvent.getSource());

        internalHandleCloudEvent(cloudEvent, payload);
    }
```

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/messaging/incoming/ExplainabilityResultConsumer.java [hunk 2]
```diff
@@ -46,17 +44,17 @@
...
-
     private ExplainabilityResultConsumer() {
         //CDI proxy
     }
 
+    @Inject
+    ObjectMapper mapper;
+
     @Inject
     public ExplainabilityResultConsumer(TrustyService service) {
```
Comment (danielezonca) [inline trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/messaging/incoming/ExplainabilityResultConsumer.java]: Please do not mix injection on field and on constructor: move all the injection to constructor

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/messaging/incoming/ModelEventConsumer.java [hunk 3]
```diff
@@ -33,15 +31,15 @@
...
+public class ModelEventConsumer extends BaseEventConsumer {
 
     private  ModelEventConsumer() {
         //CDI proxy
     }
 
+    @Inject
+    ObjectMapper mapper;
+
     @Inject
```
Comment (danielezonca) [inline trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/messaging/incoming/ModelEventConsumer.java]: Please do not mix injection on field and on constructor: move all the injection to constructor

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/messaging/incoming/TraceEventConsumer.java [hunk 4]
```diff
@@ -33,15 +31,15 @@
...
+public class TraceEventConsumer extends BaseEventConsumer {
 
     private TraceEventConsumer() {
         //CDI proxy
     }
 
+    @Inject
+    ObjectMapper mapper;
+
     @Inject
```
Comment (danielezonca) [inline trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/messaging/incoming/TraceEventConsumer.java]: Please do not mix injection on field and on constructor: move all the injection to constructor

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/messaging/BaseEventConsumer.java (hunk 1 above)
Comment (ruromero) [reply trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/messaging/BaseEventConsumer.java]: Done, thanks for the suggestion

**Review by danielezonca — APPROVED** [review]
Thanks for your PR, approving 👍

**Comment by ricardozanini** [comment]
@danielezonca I also need this PR merged, but it looks like it's failing in the CI. I was incapable to see why, tough day. Can you take a look tomorrow? After a good night of sleep I'll look also.

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
@ruromero 
Can you please have a look to failed tests?
`testEvaluateTrafficViolation – org.kie.dmn.kogito.quarkus.tracing.LoanEligibilityTest` fails with 
```bash
2020-09-16 04:12:54,840 INFO [io.sma.rea.mes.provider] (main) SRMSG00209: Initializing mediators
2020-09-16 04:12:54,918 ERROR [io.sma.rea.mes.provider] (main) SRMSG00212: Unable to initialize mediator: org.kie.kogito.tracing.decision.QuarkusModelEventEmitter#getEventPublisher: java.lang.NoClassDefFoundError: io/cloudevents/rw/CloudEventAttributesWriter
at io.cloudevents.jackson.JsonFormat.getCloudEventJacksonModule(JsonFormat.java:102)
```

`org.acme.travel.CloudEventListenerTest – org.acme.travel.CloudEventListenerTest` fails with
```bash
[error]: Build step org.kie.kogito.quarkus.deployment.KogitoAssetsProcessor#generateModel threw an exception: java.lang.IllegalStateException: src/main/java/org/kie/kogito/app/CloudEventListenerResource.java (7:40) : cannot find symbol
symbol: class CloudEventConverter
location: package org.kie.kogito.events.knative.ce
src/main/java/org/kie/kogito/app/CloudEventListenerResource.java (27:13) : cannot find symbol
symbol: class RestEasyHttpRequestConverter
location: class org.kie.kogito.app.CloudEventListenerResource
src/main/java/org/kie/kogito/app/CloudEventListenerResource.java (27:69) : cannot find symbol
symbol: class RestEasyHttpRequestConverter
```
```

**Comment by ruromero** [comment]
@danielezonca yes I'm still working on it. I'm having trouble with the objectMapper.

**Comment by ruromero** [comment]
jenkins retest this

**Review by ricardozanini — CHANGES_REQUESTED** [review]
LGTM @kostola @manstis do you mind taking a look? I should merge it tomorrow if nothing goes wrong.

**Review by ricardozanini — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**